### ISSUE TITLE
Use immutables for flows

### ISSUE THREAD

**Issue by davidliu** [livekit/client-sdk-android#101]
State flows won't refire for internal changes within objects. Need to create immutable versions for flows.

**Comment by davidliu** [comment]
@AchrafAmil note that the track isn't replaced when camera is flipped; the LocalVideoTrack instance remains the same.

**Comment by AchrafAmil** [comment]
Yes! Until it's made immutable and would therefore inevitably be a new instance.
